src/drawing/backend_impl/svg.rs

```diff
@@ -253,9 +255,16 @@ impl<'a> DrawingBackend for SVGBackend<'a> {
         let x0 = pos.0 + offset.0;
         let y0 = pos.1 + offset.1;
 
+        let max_x = (layout.1).0;
+        let (dx, anchor) = match style.alignment {
+            TextAlignment::Left => (0, "start"),
```

Comment:
I feel this `TextAlignment` is more confusing than `TextAnchor`. How do you think about this ? And document this for the fact not all backends agree with fontkits' text size estimation, thus we need to define a invariant point 
Since text alignment may also means we want to align the text on the page.

Reply:
TextAnchor sounds good, I like it (I still would prefer to have the rectangle and text alignment within the rectangle, but it's a completely different story). I'll update it.

Reply:
But `TextAnchor` is confusing as well. So if you think it's not confusing at all, just make sure documentation describes what it's about. :)

Reply:
Okay, what about we leave TextAlignment as-is, but rather change the draw_text API to use rectangle instead of (x, y), so the alignment is going to be used properly? I've verified the code, looks like the current implementations could be improved:
- cairo: it's better to use their `text_extents` method to get the proper "text inked" rectangle;
- svg and canvas: don't need "text inked" rectangle calculations at all;
- other backends use `font.draw` method that calculates the rectangle again (!).

However, it's quite a change, and I'd prefer to submit another PR if you don't mind. Also, if you okay with that, do you want full alignments implemented (including vertical), or just horizontal would be enough for now? Surely, we assume one-line text only, multiline text block would require even more alignment options, and could be out of scope of the project completely.

Reply:
Sure, as you might realized, the text positioning code is some legacy convention which can be dated back to the first 300 lines of code in this crate and it causing problems for sure. 
You are more than welcome to make any improvement on this area.  :)

------

And for multi-line text, there's an element for that, called `MultiLineText`, which performing the text layout when it's drawn to backend.